vm/src/obj/objsequence.rs

```diff
@@ -137,6 +137,107 @@ impl<T: Clone> PySliceableSequence for Vec<T> {
     }
 }
 
+pub trait PySliceableSequenceMut: PySliceableSequence {
+    //fn set_slice<T: PySliceableSequence>(&self, range: Range<usize>, seq: T) -> Self;
+    //fn set_slice_reverse<T: PySliceableSequence>(&self, range: Range<usize>, seq: T) -> Self;
+    //fn set_stepped_slice<T: PySliceableSequence>(&self, range: Range<usize>, step: usize, seq: T) -> Self;
+    //fn set_stepped_slice_reverse<T: PySliceableSequence>(&self, range: Range<usize>, step: usize, seq: T) -> Self;
+
+    fn del_slice(&mut self, range: Range<usize>);
+    fn del_stepped_slice(&mut self, range: Range<usize>, step: usize);
+    fn del_index(&mut self, index: usize);
+
+    fn del_slice_items(&mut self, vm: &VirtualMachine, slice: &PyObjectRef) -> PyResult {
```

Comment:
if the only possible type for `slice` is a `PySlice`, let's convert the reference to `PySliceRef` in `del_item`, and pass it here. Can remove the `payload => panic!` branch then.

Reply:
a good point, done with the latest commit.